docs/user-manual/core-tools.md

```diff
@@ -219,12 +219,354 @@ As c++ is a strongly typed language, the signal template parameters signature ha
+    <counter value="$FOR{count}"/>
+  </for>
+</dqm4hep>
+```
+
+will be replaced by:
+```xml
+<dqm4hep>
+  <!-- The current value of the counter is 0 -->
+  <counter value="0"/>
+  <!-- The current value of the counter is 1 -->
+  <counter value="1"/>
+  <!-- The current value of the counter is 2 -->
+  <counter value="2"/>
+  <!-- The current value of the counter is 3 -->
+  <counter value="3"/>
+</dqm4hep>
+```
+
+By construction, it is also possible the run nested for loops by changing the loop id:
+
+```xml
+<dqm4hep>
+  <for id="x" begin="0" end="2" increment="1">
+    <for id="y" begin="0" end="2" increment="1">
+      <entry x="$FOR{x}" y="$FOR{y}"/>
+    </for>
+  </for>
+</dqm4hep>
+```
+
+will be replaced by:
+
+```xml
+<dqm4hep>
+  <entry x="0" y="0"/>
+  <entry x="0" y="1"/>
+  <entry x="0" y="2"/>
+  <entry x="1" y="0"/>
+  <entry x="1" y="1"/>
+  <entry x="1" y="2"/>
+  <entry x="2" y="0"/>
+  <entry x="2" y="1"/>
+  <entry x="2" y="2"/>
+</dqm4hep>
+```
+
+The `<for>` element attributes possible values are summurized in table below. 
+ 
+Attribute | Type    | Optional ? | Default value
+--------- | ------- | ---------- | -------------
+id        | string  | false      | - 
+begin     | integer | false      | - 
+end       | integer | false      | -
+increment | integer | true       | 1
+
+## XML parsing ordering
+
+The XML file is processed in the given order:
+
+- Includes resolution
+- Constants and environment variables
+- Database connections and parameter select
+- For loops
+
+This has the following effects:
+
+1. Constants and environment variables can be used in database connections and for loops
+2. As includes are processed before constants, the `ref` attribute of the `<include>` element can not refer to any constant. In this case the constant will not replaced and the file will not been included.
+3. As constants are processed after includes, it possible to define constants in an included file. The `<constants>` section in the included file has to be defined in the root element and the `<include>` element in the main file has to be defined in the root element.
+
+## Using the XMLParser class
+
+The `XMLParser` class is part of the DQMCore distribution and can be included using the directive:
+
+```cpp
+#include <dqm4hep/XMLParser.h>
+using XMLParser = dqm4hep::core::XMLParser;
+```
+
+To parse an XML file, use the method `XMLParser::parse()`:
+
+```cpp
+XMLParser parser;
+try {
+  parser.parse("myfile.xml");
+}
+catch(StatusCodeException &exception) {
+  dqm_error( "Caught StatusCodeException while parsing XML file: {0}", exception.toString() );
+}
+```
+
+The XML document itself can be accessed with the method `XMLParser::document()`:
+
+```cpp
+auto document = parser.document();
+```
+
+All of the features listed above are by default active but can be de-activated using dedicated method. The example below shows how to de-activate all the features listed above. 
```

Comment:
All of the features listed above are by default active but can be de-activated using dedicated **methods**.
**A warning is sent to the logger when a feature is manually deactivated**.